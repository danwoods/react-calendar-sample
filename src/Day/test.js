import React from 'react'
import { Day } from './index.js'
import renderer from 'react-test-renderer'

const date = new Date(2000, 0, 1)

describe('Day component', () => {
  test('Basic Day', () => {
    const component = renderer.create(<Day date={date} />)
    let tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('Day with children', () => {
    const component = renderer.create(<Day date={date}>{'kids!'}</Day>)
    let tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('Day with full props', () => {
    const component = renderer.create(
      <Day
        date={date}
        contentClassName={'ccn'}
        dateLabelClassName={'dlcn'}
        onClick={() => alert('clicked!')}
        rootClassName={'rcn'}
      >
        {'kids!'}
      </Day>
    )
    let tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
