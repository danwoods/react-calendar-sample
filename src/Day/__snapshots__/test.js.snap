// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day component Basic Day 1`] = `
<div
  className="TEST-day-root "
  title="January 1, 2000"
>
  <div
    className="TEST-day-content "
  >
    <h6
      className="TEST-day-date-label "
    >
      1
    </h6>
  </div>
</div>
`;

exports[`Day component Day with children 1`] = `
<div
  className="TEST-day-root "
  title="January 1, 2000"
>
  <div
    className="TEST-day-content "
  >
    <h6
      className="TEST-day-date-label "
    >
      1
    </h6>
    kids!
  </div>
</div>
`;

exports[`Day component Day with full props 1`] = `
<div
  className="TEST-day-root rcn"
  onClick={[Function]}
  title="January 1, 2000"
>
  <div
    className="TEST-day-content ccn"
  >
    <h6
      className="TEST-day-date-label dlcn"
    >
      1
    </h6>
    kids!
  </div>
</div>
`;
